app/repository/hbase/HBaseRestRepository.scala

```diff
@@ -113,62 +115,89 @@ class HBaseRestRepository @Inject() (
     }
   }
 
+  override def createOrReplace(table: String, rowKey: RowKey, field: Field): Future[CreateOrReplaceResult] = {
+    val url = HBase.rowKeyUrl(withBase = config.baseUrl, config.namespace, table, rowKey)
+    logger.info(s"Requesting creation/replacement of [${field._1}] at [$url] via HBase REST.")
+    asCreateOrReplaceResult(putEdit(url, SingleValue(rowKey, field)))
+  }
+
+  override def updateField(table: String, rowKey: RowKey, checkField: Field, updateField: Field): Future[OptimisticEditResult] =
+    ifExists(table, rowKey, columnOf(checkField).family) { _ =>
+      val url = HBase.checkedPutUrl(withBase = config.baseUrl, config.namespace, table, rowKey)
+      logger.info(s"Requesting update of [${updateField._1}] at [$url] via HBase REST.")
+      doCheckedEdit(url, CheckAndUpdate(rowKey, checkField, updateField))
+    }
+
   /*
-   * A HBase checkAndUpdate operation returns 304: Not Modified when:
-   * - the "check" is not satisfied (ie. the value has been changed by another user)
-   * - there is no such row with the target rowKey
+   * Note that if the target column does not exist, we take no action and report a successful deletion.
+   * This is the correct behaviour for an idempotent delete operation, and ensures that the operation can be retried
+   * in the event of any failure.
+   * However, it was probably a mistake to handle this here.  The client request is actually a patch containing
+   * test & remove operations.  If the test condition does not hold, the patch operation should be aborted.  In
+   * focusing on the delete operation we have failed to correctly adhere to the expected semantics of patch.
+   */
+  override def deleteField(table: String, rowKey: RowKey, checkField: Field, columnName: Column): Future[OptimisticEditResult] =
+    ifExists(table, rowKey, columnName.family) { row =>
+      if (row.fields.contains(columnName.qualifier)) checkAndDelete(table, rowKey, checkField, columnName)
+      else Future.successful(EditApplied)
+    }
+
+  private def checkAndDelete(table: String, rowKey: RowKey, checkField: Field, columnName: Column): Future[OptimisticEditResult] = {
```

Comment:
As I see it, the following happens within `checkAndDelete`: `doCheckedEdit` -> `asOptimisticEditResult`
It is not obvious that anything is deleted, maybe `doCheckedEdit` should be something like `doCheckedEditAndDelete`?

Reply:
```doCheckedEdit``` is a general-purpose method that sends a ```PUT``` request to HBase and translates the status code.  It is used by both delete & update operations (note that HBase uses a PUT to do a checked delete operation not a DELETE).

The main thing that distinguishes the operations is the url - which are obtained from ```checkedDeleteUrl``` or ```checkedPutUrl```.

I can rename the url variables so that you get something like
```doCheckedEdit(deleteUrl, ...)``` if that helps?

Reply:
That sounds more intuitive!